datafusion/sql/src/planner.rs

```diff
@@ -1720,22 +1721,22 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 ],
             }),
 
-            SQLExpr::Value(Value::Interval {
+            SQLExpr::Array(arr) => self.sql_array_literal(arr.elem, schema),
+
+            SQLExpr::Interval {
```

Comment:
Intervals are no longer literals (`Value`) but can have exprs so the AST was updated to match in https://github.com/sqlparser-rs/sqlparser-rs/pull/609